Xamarin.Forms.Core/Xaml/IResourcesLoader.cs

```diff
@@ -7,7 +7,7 @@ namespace Xamarin.Forms
 {
 	interface IResourcesLoader
 	{
-		T CreateFromResource<T>(string resourcePath, Assembly assembly, IXmlLineInfo lineInfo) where T : new();
+		T CreateFromResource<T>(string resourcePath, Assembly assembly, IXmlLineInfo lineInfo, ResourceDictionary parentRD) where T : new();
```

Comment:
Looking at this particular member, I think it is time to rename to a more "resource dictionary" name - maybe `CreateResourceDictionary` or `LoadResourceDictionary`. Right now, we are just using it to load a dictionary - and we have code in there just for dictionaries. If we rename and go full in, we can remove the generics and the casts:

```csharp
ResourceDictionary CreateResourceDictionary(string resourcePath, Assembly assembly, IXmlLineInfo lineInfo, ResourceDictionary parent);
```

```csharp
public ResourceDictionary CreateResourceDictionary(string resourcePath, Assembly assembly, IXmlLineInfo lineInfo, ResourceDictionary parent) {
    var dict = new ResourceDictionary();
    dict.Parent = parent;
    return dict.LoadFromXaml(GetResource(resourcePath, assembly, lineInfo));
}
```

Reply:
Another alternative is to add a _new_ member that is dictionary specific, and keep the old one:

```csharp
public T CreateFromResource <T>(string resourcePath, Assembly assembly, IXmlLineInfo lineInfo) {
    var res = new T();
    return res.LoadFromXaml(GetResource(resourcePath, assembly, lineInfo));
}
```

Reply:
I third alternative is to go with option 2 in this file (keep generic), but do the `ResourceDictionary` specific bits in the `ResourceDictionary` type (instead of calling this method to load, just do it there). This is also a good option to be consistent with what we do with the css loading: https://github.com/xamarin/Xamarin.Forms/blob/f2948f1f0cf9829fe6ddafc60df2ac06a6a59ed0/Xamarin.Forms.Xaml/MarkupExtensions/StyleSheetExtension.cs#L35-L36

Reply:
Of these 3 options - I vote 3 as I see no value with having a generic method that is used in only 1 place for a specific purpose in an unrelated file. The resource loader should be responsible for loading the actual files - the dictionary should be responsible for "parsing" those files (same with the css)